### ISSUE TITLE
New rule: flag time.Sleep() in test files

### ISSUE THREAD

**Issue by Acollie** [mgechev/revive#1754]
**Is your feature request related to a problem? Please describe.**
I get frustrated when reviewing PRs and a test has a `time.Sleep()` in it it's usually a sign the test needs to be refactored.

Example:
```go
func TestWorkerProcessesJob(t *testing.T) {
    w.Start()
    time.Sleep(200 * time.Millisecond) // waiting for the worker to finish
    assert.True(t, w.Done())
}
```

**Describe the solution you'd like**
A simple rule which fails when a test has a `time.Sleep()`.

**Additional context**
[This is a well-documented testing anti-pattern: see Martin Fowler's views here](https://martinfowler.com/articles/nonDeterminism.html)

**Comment by ccoVeille** [comment]
This can be achieve via forbidigo, available via golangci-lint.

This is how I report them usually.

I don't see a need for time.Sleep in code usually using a context.WithTimeOut for something that expires, or a time.Ticker when time.Sleep is used in a loop.

Context is often better especially since the addition of t.Context.

But for your need on reporting in test files, forbidigo plus an exclusion on non-test files could do the job.

I'm unsure there is a need for this in revive, but I'm curious about other people feedbacks

**Comment by ccoVeille** [comment]
Thar said there is a niche for well know anti pattern in the Go linting ecosystem where a lot of linter where refused from golangci-lint because of the "it can be done via forbidigo/ruleguard" explanation

Which means there is nothing in the go ecosystem to report them, because there is a no "it's dangerous to go alone, take this forbidigo sets you can use"

So maybe it's OK to start adding them in revive 
.

**Comment by ccoVeille** [comment]
Found back this

- https://github.com/golangci/golangci-lint/discussions/4753

**Comment by Acollie** [comment]
Thanks for the response! @ccoVeille  Totally agree it's a big anti-pattern.

  I personally think we should include this as part of revive rather than relying on another package. forbidigo is more of a regex-style solution, whereas I think a more robust AST-based approach is the right way
  to catch this. The goal of revive being a batteries-included linter points to including it.

**Comment by alexandear** [comment]
Thanks for reporting.

I agree with @ccoVeille. We can use the `forbidigo` to resolve your problem, either a [standalone](https://github.com/ashanbrown/forbidigo) or [inside the golangci-lint](https://golangci-lint.run/docs/linters/configuration/#forbidigo).

Golangci-lint config could be:

```yaml
version: "2"

linters:
  enable:
    - forbidigo
  settings:
    forbidigo:
      analyze-types: true
      forbid:
        - pattern: ^time\.Sleep.*$
          msg: Do not use time.Sleep.
  exclusions:
    rules:
      - path-except: (.+)_test\.go
        linters:
          - forbidigo
```
